### DISCUSSION TITLE
Using Zeiss's diagrams in PDF spec sheets instead of measuring by hand?

### DISCUSSION THREAD

**Discussion by qumuq-til** [lensfun/lensfun#1624] [General]
For example, https://www.zeiss.de/content/dam/consumer-products/downloads/photography/datasheets/en/milvus-lenses/datasheet-zeiss-milvus-2821.pdf right on page 2 there's a nice graph that shows how the lens distorts. They publish similar specs for the entire Milvus, Otus, Batis, Loxia and even Touit ranges. Possibly even older lenses as well. Can someone say that
1. It makes sense to directly input this data into lensfun correction database instead of trying to shoot rectangles with each one of them and
2. Share an algorithm (manual) to convert the data in the graphs to datapoints in lensfun's xml files

**Comment by uchrisu** [comment]
It should be possible but it should be checked against real images, at least for a few lenses I think.

However, the translation from the graphs to pairs of numbers has to be done by hand then, i.e., somebody would have to create a list with pairs of image height and distortion for each lens. 

For Vignetting there are only two apertures. Too few for our purpose and there is no TCA graph. So probably its only good for distortion, what will already be quite helpful.

**Comment by qumuq-til** [comment]
So then it would be a good start. Can you show me an example of a workflow, maybe from one of their graphs so I could compile the necessary data, maybe even a complete XML snippet for each lens? I can translate the graph to numbers but not sure where I should put the sampling points (or if I need to derive curve formulas from that) and what data lensfun needs exactly. Milvus 21 would be a good example due to its non-trivial distortion profile.
